AppleALC/kern_alc.cpp

```diff
@@ -279,6 +279,37 @@ IOService *AlcEnabler::gfxProbe(IOService *ctrl, IOService *provider, SInt32 *sc
 	return FunctionCast(gfxProbe, callbackAlc->orgGfxProbe)(ctrl, provider, score);
 }
 
+bool AlcEnabler::AppleHDAController_start(IOService* service, IOService* provider)
+{
+	uint32_t delay = 0;
+	if (PE_parse_boot_argn("alcdelay", &delay, sizeof(delay))) {
+		DBGLOG("alc", "found alc-delay override %u", delay);
+		provider->setProperty("alc-delay", &delay, sizeof(delay));
+	} else {
+		if (WIOKit::getOSDataValue(provider, "alc-delay", delay))
+			DBGLOG("alc", "found normal alc-delay %u", delay);
+	}
+	
+	if (delay > 3000) {
+		SYSLOG("alc", "alc delay cannot exceed 3000 ms, ignore it");
+		delay = 0;
+	}
+		
+	if (delay != 0) {
+		DBGLOG("alc", "delay AppleHDAController::start for %d ms", delay);
+		IOSleep(delay);
+	}
+	return FunctionCast(AppleHDAController_start, callbackAlc->orgAppleHDAController_start)(service, provider);
+}
+
+IOReturn AlcEnabler::IOHDACodecDevice_executeVerb(void *that, uint16_t a1, uint16_t a2, uint16_t a3, unsigned int *a4, bool a5)
```

Comment:
I am not positive this route is needed for this latch, maybe remove it?

Reply:
I would prefer to keep this route, since we can easily detect any issues with executeVerb in the future. It also can be useful for debugging custom verbs added to some codecs. If you know any another way to such issues in logs, I will remove this.

Reply:
Hm, in this case it should be guarded with ifdefs to be debug-only, so that we do not reduce the performance of release builds.